### ISSUE TITLE
11 of 12 words are known and the ordering is unknown

### ISSUE THREAD

**Issue by counterstrike777** [3rdIteration/btcrecover#20]
Thank you for adding discrambling method of 12 seed words.

If I have 11 words of bip39 mnemonic, 1 word is unknown and the ordering of all is unknown. Is it possible to use it?

**Comment by 3rdIteration** [comment]
Yep, you can just follow the syntax here: https://github.com/3rdIteration/btcrecover/blob/master/docs/Usage_Examples/2020-05-02_Descrambling_a_12_word_seed/fixedwords_bip39.txt

Basically what you will want is to have the 1st 11 rows of the file without any positional markers, like the rows 7-11 of that file. The last row of the file would then contain all 2048 BIP39 words in the same way that the example shows all BIP39 words starting with "B".

Basically it will have 2048 * 11! = 81749606400 possible seeds to work through, so on an average PC, it could take about a month to run. (If you run it on a 48 core Linode you could knock it over in about 7 days max if you skip the counting step and go with the --no-eta option)

**Comment by 3rdIteration** [comment]
OP question answered with no followup questions for 10 days.

**Comment by jonathancross** [comment]
@3rdIteration  I was wondering, have you explored an option to skip all of the wallet creation, address checking, etc for such a situation?  I am wondering if simply re-arranging the words until you get a valid BIP39 checksum would be faster?

**Comment by 3rdIteration** [comment]
That's actually sort of what already happens. Basically, the hashing (the slow bit) to produce a wallet master key from a seed phrase only happens when a checksum is valid.

I just did a quick run of 50,000 seeds to generate and basically:
For BIP39: Seeds Hashed: 3323 Seeds Checksum Failed: 50285
For Electrum: Seeds Hashed: 411 Seeds Checksum Failed: 52031

The more robust check-summing is why BTCRecover can hash Electrum seeds faster than BIP39 seeds... (Though not 10x faster, as there is still some CPU overhead associated with check-summing)

Essentially, the checksums in BIP39 seeds aren't super robust and only provide a very basic level of protection against making a typo... (Compared to how Bitcoin addresse checksum where the chance of having an error that still checksums is like 1 in 4 billion)

**Comment by jonathancross** [comment]
Thanks for doing this.

> For BIP39: Seeds Hashed: 3323 Seeds Checksum Failed: 50285

Sorry, I am missing something...

* 3223 seed phrases that passed the checksum
* 50285 seed phrases that failed to checksum

...in a run of only 50,000?

**Comment by 3rdIteration** [comment]
Yea basically.

It's something I will document more with the GPU acceleration stuff, as it actually makes a really big difference there and is important for optimising workload size.

Basically for a 12 word seed, the checksumming leaves 1/16 of the candidate phrases to check, for a 24 it's basically 1/256, Electrum needs it's about 1/125. For CPU processing, the checksumming is much faster than computing and checking a seed, but for GPU it evens it out to the extent that for a 24 word seed, it actually bottlenecks due to the checksumming.

The opencl branch will have a new argument, similar to listseeds, that only writes out seeds with a valid checksum. This will be useful in a few ways, bit will also illustrate how many seeds get dropped for checksumming really clearly.

**Comment by jonathancross** [comment]
Thanks for explaining, but I'm confused because it seems the number of failures (50285) was larger than the total number of seeds you tested ("a run").

**Comment by 3rdIteration** [comment]
Ah right, yes that was me just terminating it manually at around 50,000 :)
